File: API/IAMTtradingWA/Models/PurchaseOrdersVendors/PurchaseOrdersVendors.cs
using System;

namespace IAMTradingWA.Models
{
    public class PurchaseOrderVendor
    {
        public int POVId { get; set; }
        public int QHeaderId { get; set; }
        public int VendorId { get; set; }
        public string POVType { get; set; }
        public decimal POVAmount { get; set; }
        public decimal POVAmountNB { get; set; }
        public DateTime POVDate { get; set; }
        public decimal POVCurrencyRate { get; set; }
        public int? POVParentId { get; set; }
        public string POVPaymentNumber { get; set; }
        public string POVCreatedBy { get; set; }
        public DateTime POVCreatedDate { get; set; }
        public string POVModifiedBy { get; set; }
        public Nullable<DateTime> POVModifiedDate { get; set; }
        public string x_VendorName { get; set; }
        public string x_QHeaderReference { get; set; }
        public Nullable<decimal> x_InvoiceVendorBalance { get; set; }
        public Nullable<decimal> x_InvoiceVendorBalanceNB { get; set; }
        public Nullable<decimal> x_TotalBalance { get; set; }
        public bool x_Selected { get; set; }
        public int CCategoryId { get; set; }
    }

    public class POVendorsDetail
    {
        public int POVDetailId { get; set; }
        public int POVId { get; set; }
        public decimal POVDetailAmount { get; set; }
        public decimal POVDetailAmountNB { get; set; }
        public decimal POVDetailCurrencyRate { get; set; }
        public string POVDetailPaymentNumber { get; set; }
        public string POVDetailType { get; set; }
        public string POVDetailCreatedBy { get; set; }
        public DateTime POVDetailCreatedDate { get; set; }
        public string POVDetailModifiedBy { get; set; }
        public Nullable<DateTime> POVDetailModifiedDate { get; set; }
        public DateTime POVDetailDate { get; set; }
        public int PayVendorId { get; set; }
    }
}

File: WEB/app/app/model/PurchaseOrdersVendors.js
Ext.define('IAMTrading.model.PurchaseOrdersVendors', {
    extend: 'Ext.data.Model',
    idProperty: 'POVId',
    autoLoad: false,

    fields: [
    { name:'POVId', type:'int', defaultValue: null },
    { name:'QHeaderId', type:'int', defaultValue: null },
    { name:'VendorId', type:'int', useNull: true, defaultValue: null },
    { name:'POVType', type:'string' },
    { name:'POVAmount', type:'float', defaultValue: null },
    { name:'POVAmountNB', type:'float', defaultValue: null },
    { name:'POVDate', type:'date', defaultValue: null },
    { name:'POVCurrencyRate', type:'float', defaultValue: null },
    { name:'POVParentId', type:'int', useNull: true, defaultValue: null },
    { name:'POVPaymentNumber', type:'string', useNull: true, defaultValue: null },
    { name:'POVCreatedBy', type:'string', useNull: true, defaultValue: IAMTrading.GlobalSettings.getCurrentUserId() },
    { name:'POVCreatedDate', type:'date', defaultValue: null },
    { name:'POVModifiedBy', type:'string', useNull: true, defaultValue: null },
    { name:'POVModifiedDate', type:'date', useNull: true, defaultValue: null },
    { name:'x_VendorName', type:'string', useNull: true, defaultValue: null },
    { name:'x_QHeaderReference', type:'string', useNull: true, defaultValue: null },
    { name:'x_InvoiceVendorBalance', type:'float', useNull: true, defaultValue: null },
    { name:'x_InvoiceVendorBalanceNB', type:'float', useNull: true, defaultValue: null },
    { name:'x_TotalBalance', type:'float', useNull: true, defaultValue: null },
    { name:'x_Selected', type:'boolean', defaultValue: false }
    ],

    proxy:{
        type:'rest',
        url: IAMTrading.GlobalSettings.webApiPath + 'PurchaseOrdersVendors',
        headers: {
           'Authorization-Token': IAMTrading.GlobalSettings.tokenAuth
        },
        reader:{
            type:'json',
            root:'data',
            totalProperty: 'total',
            successProperty: 'success',
            messageProperty: 'message',
            idProperty: 'POVId'
        },
        writer: {
            writeAllFields: false
        },
        afterRequest: function (request, success) {
            var silentMode = this.getSilentMode();

            if (request.action == 'read') {
                //this.readCallback(request);
            }
            else if (request.action == 'create') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not saved!!!");
                    Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Saved Successfully!!!");
                }
            }
            else if (request.action == 'update') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not saved!!!");
                    Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Saved Successfully!!!");
                }
            }
            else if (request.action == 'destroy') {
                if (!request.operation.success)
                {
                    Ext.popupMsg.msg("Warning", "Record was not Deleted");
                    //Ext.global.console.warn(request.proxy.reader.jsonData.message);
                } else {
                    if(!silentMode)
                    Ext.popupMsg.msg("Success","Deleted Successfully");
                }
            }

            this.setSilentMode(false);
        }
    },


});